### DISCUSSION TITLE
Cron jobs triggered on every deployment

### DISCUSSION THREAD

**Discussion by flosrn** [vercel/next.js#51509] [Help]
### Summary

Hello,

I'm using Vercel to host my Next.js application and I have set up a couple of cron jobs as per the [official documentation](https://vercel.com/docs/cron-jobs).

Here is my `vercel.json` configuration for the cron jobs:

```json
{
  "crons": [
    {
      "path": "/api/cron/reset-free-users-credits",
      "schedule": "0 8 * * *"
    },
  ]
}
```

And here my route `/app/api/cron/reset-free-users-credits.ts`
```ts
import { NextResponse } from "next/server";
import { prisma } from "~/server/db/prisma";

export async function GET() {
  // reset credits to 5 for all users in FREE plan
  try {
    await prisma.user.updateMany({
      where: { plan: "FREE" },
      data: { credits: 5 },
    });
    console.log(`FREE plan cron success at ${new Date().toLocaleString()}`);
    return NextResponse.json({ status: 200 });
  } catch (error: unknown) {
    console.log("cron error:", error);
    return NextResponse.json({ status: 500 });
  }
}

```

As per my understanding, these cron jobs should only trigger based on the schedule specified. However, I've noticed that these cron jobs are being executed every time I deploy my application on Vercel.

I tried searching for information regarding this issue, but I couldn't find anything in the official documentation or in community discussions that suggests that cron jobs should run on every deployment.

Could you please help me understand why this is happening and how I can prevent the cron jobs from running on every deployment? Any guidance would be greatly appreciated.

Thank you for your time and assistance!

**Comment by flosrn** [comment]
I taken a look on the Vercel build and it seems that Vercel execute my cron on every build

![CleanShot 2023-06-19 at 18 47 42@2x](https://github.com/vercel/next.js/assets/12709355/97dc4cf7-9609-448b-af2e-3ad6aa617bf1)

I don't understand why

**Comment by flosrn** [comment]
I moved my `cron` folder from `app/api` directory to `pages/api` directory and now it work, no more cron triggered on every deployment, is this a bug?

**Comment by henriquepcm** [comment]
I had the same issue here. I got this fixed keeping the "api" folder in "app", and added `export const dynamic = "force-dynamic";` to force the page to be treated as dynamic and not render at building time.

Context: https://nextjs.org/docs/messages/app-static-to-dynamic-error
